File: Chapter09/standardBrownianMotion.clj
; Written by Raju
; Run using following command
; clojure standardBrownianMotion.clj
;Standard Brownian Motion in Clojure

 (defn brownianPathGenerator [x] ;; Line One 
  (def rndm (java.util.Random.)) ;; Line Two
  (def y (.nextGaussian rndm)) ;; Line Three
 (lazy-seq (cons x (brownianPathGenerator (+ x y))))) ;; Line Four 

(println  (take 10 (brownianPathGenerator 0)))

File: Chapter09/standardBrownianMotion.scala
// Written by Raju
// Run it using following command
//  scala standardBrownianMotion.scala

import Stream._

def brownianPathGenerator( x : Double) : Stream[Double] = {
        val y = scala.util.Random.nextGaussian
        cons(x, brownianPathGenerator(x+y))
     }

val brownianPath = brownianPathGenerator( 0 ).take(10)
println("Printing 10 elements of a Standard Brownian Path")
println("------------------------------------------------")
brownianPath.toList.foreach(println)
